iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/PhaseShifterShiftTap.java

```diff
@@ -55,4 +55,23 @@ private void adjustTapPosition(PhaseTapChanger phaseTapChanger) {
         phaseTapChanger.setTapPosition(Math.min(Math.max(phaseTapChanger.getTapPosition() + tapDelta,
                 phaseTapChanger.getLowTapPosition()), phaseTapChanger.getHighTapPosition()));
     }
+
+    @Override
+    public NetworkModificationImpact hasImpactOnNetwork(Network network) {
+        impact = DEFAULT_IMPACT;
+        TwoWindingsTransformer phaseShifter = network.getTwoWindingsTransformer(phaseShifterId);
+        if (phaseShifter == null || !phaseShifter.hasPhaseTapChanger()) {
+            impact = NetworkModificationImpact.CANNOT_BE_APPLIED;
```

Comment:
do we also want to verify that the tap is in the PST's actual tap range?

Reply:
In the present case it would be impossible since tapPosition is computed as: `tapPosition = Math.min(Math.max(phaseTapChanger.getTapPosition() + tapDelta, phaseTapChanger.getLowTapPosition()), phaseTapChanger.getHighTapPosition())` so the tap is always in the PST's actual tap range

Reply:
I wonder if it would not be cleaner to remove min/max and return a CANNOT_BE_APPLIED. Because from a user perspective, they won't know that they set, for example, a tap position that is too high, and they'll think that the tap position was accepted as is.
What do you think?

Reply:
maybe that's not the idea behind this action which is an iterative one. if that's the case then it's ok for me

Reply:
Not sure about the idea behind this network modification.
But if we want to change how it works, it has to be done in another PR since the present one is only to add the checks according to how the network modifications currently work.